libraries/stdlib/stdlib_ng.mtlx

```diff
@@ -265,38 +265,52 @@
     <normalize name="N_norm_float" type="vector3">
       <input name="in" type="vector3" interfacename="normal" />
     </normalize>
-    <dotproduct name="N_dotX_float" type="float">
-      <input name="in1" type="vector3" nodename="N_norm_float" />
-      <input name="in2" type="vector3" value="1.0, 0.0, 0.0" />
-    </dotproduct>
-    <dotproduct name="N_dotY_float" type="float">
-      <input name="in1" type="vector3" nodename="N_norm_float" />
-      <input name="in2" type="vector3" value="0.0, 1.0, 0.0" />
-    </dotproduct>
-    <dotproduct name="N_dotZ_float" type="float">
-      <input name="in1" type="vector3" nodename="N_norm_float" />
-      <input name="in2" type="vector3" value="0.0, 0.0, 1.0" />
-    </dotproduct>
-    <absval name="N_blendX_float" type="float">
-      <input name="in" type="float" nodename="N_dotX_float" />
-    </absval>
-    <absval name="N_blendY_float" type="float">
-      <input name="in" type="float" nodename="N_dotY_float" />
-    </absval>
-    <absval name="N_blendZ_float" type="float">
-      <input name="in" type="float" nodename="N_dotZ_float" />
+    <absval name="N_absN" type="vector3">
+      <input name="in" type="vector3" nodename="N_norm_float" />
     </absval>
+    <dotproduct name="N_dotN" type="float">
+      <input name="in1" type="vector3" nodename="N_absN" />
+      <input name="in2" type="vector3" value="1.0, 1.0, 1.0" />
+    </dotproduct>
+    <subtract name="N_normalizeWeights" type="vector3">
+      <input name="in1" type="vector3" nodename="N_absN" />
+      <input name="in2" type="float" nodename="N_dotN" />
+    </subtract>
+    <clamp name="N_clampForPrecision" type="float">
+      <input name="in" type="float" interfacename="blend" />
+      <input name="low" type="float" value="0.03" />
+      <input name="high" type="float" value="1.0" />
+    </clamp>
+    <divide name="N_oneOverBlend" type="float">
+      <input name="in1" type="float" value="1.0" />
+      <input name="in2" type="float" nodename="N_clampForPrecision" />
+    </divide>
+    <power name="N_blendPower" type="vector3">
+      <input name="in1" type="vector3" nodename="N_normalizeWeights" />
+      <input name="in2" type="float" nodename="N_oneOverBlend" />
+    </power>
+    <dotproduct name="N_dotBlendedN" type="float">
```

Comment:
@crydalch @BrianSharpe I haven't yet dived deeply into the math of this node, but one issue I notice is that the `N_dotBlendedN` is computed but never used in the graph,  Could this perhaps be a clue as to what needs fixing?